src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -614,4 +592,27 @@ export default class MongooseAdapter {
 
     return words.join(" ");
   }
+
+  static makeIdQuery(idOrIds) {
+    const idIsValid = mongoose.Types.ObjectId.isValid;
```

Comment:
I just looked up the [source](https://github.com/mongodb/js-bson/blob/master/lib/bson/objectid.js#L241) for this, and it seems to be too lenient, i.e. `idIsValid("923456789112") === true`, even though mongoose seems to give an error when trying to use/cast `"923456789112"` as an ObjectId, because it expects the 24-character format.